### DISCUSSION TITLE
Change person name of all photos in an album

### DISCUSSION THREAD

**Discussion by tommytomatoe** [RhetTbull/osxphotos#1590] [Q&A] [answered]
First off, thanks for putting together this tool! The Discussion and Q&A has been awesome as well.

Goal: I have an album where I've added photos of a person who was mislabeled as someone else. I would like to relabel all of these photos as the right person.

How can I do this with osxphotos?

A little background: I have two people in my library who look very similar. Apple Photos sometimes labels them as "Person A" but sometimes "Person B". Before discovering osxphotos, I batch selected and clicked the "These are not Person Name" option.

**Comment by RhetTbull** [comment]
Unfortunately osxphotos cannot change / add / remove person names or faces from photos. It can read them but cannot change them. This is a limitation of the Apple Photos API. I have some ideas of how to get around this but it requires more reverse engineering work with private, undocumented APIs and is not something that I can accomplish in the near term. 

Simply changing the person associated with a face in the Photos database wouldn't be too hard but these changes would not propagate to iCloud and might cause other unintended side effects.

**Reply by tommytomatoe** [reply]
Thanks for the quick response! Totally makes sense and no rush. This is a very edge use case.

On a related note, since you're very knowledgeable of the Apple Photos database setup, perhaps you can help me with this.

Background: I accidentally merged two people while going through the "Review More Photos" interface. I said "Yes" to the wrong face, and both people merged. It was a mess.

I tried to clean it up by selecting all photos (20,000 or so) and selected the "These are Not Person Name" option. This unassigned all those faces. Then I used the "Reset Suggested People & Pets" feature. 

Over the last few days, new faces were suggested and I assigned the two people to their respective names.

That said, a lot of the photos are being assigned to a completely different person who looks similar. Thus my original question above.

New questions:

1. Does using the "These are Not Person Name" feature permanently mess up the automatic matching? 
2. Can I reset a person's "profile" somehow? 

What I'd like to avoid is having to periodically and manually reassign a face to the right name.

Thank you!

**Reply by RhetTbull** [reply, marked as answer]
> Does using the "These are Not Person Name" feature permanently mess up the automatic matching?

I don't know the actual implementation Apple is using (they are of course not very transparent about this) but in my experience, using the "These are not Person..." shouldn't mess anything up permanently. It should just tell Photos that those specific faces don't go with the person it matched them to.

> Can I reset a person's "profile" somehow?

You can remove a person in the People section of the library. Right click on the person and select "Remove this Person". That removes the person from the library and all data associated with the person. It does not remove the faces, just the person those faces are associated with. But other than I don't think there's anything else you can do to "reset" as person. Note: I just tested this in Photos by snapshotting the Photos database, removing a person, and looking at changes in the database. These are the only changes:

```sql
UPDATE ZPERSON SET Z_OPT=14, ZTYPE=-1 WHERE Z_PK=5;
UPDATE Z_PRIMARYKEY SET Z_MAX=2302 WHERE Z_ENT=16001;
UPDATE Z_PRIMARYKEY SET Z_MAX=487 WHERE Z_ENT=16002;
```

I'm guessing that `ZTYPE=-1` means the person was deleted.  Will do some testing on this later.
